components/resourceDetails/resourceSharing/addAgentButton/agentPickerModal/index.jsx

```diff
@@ -261,63 +295,91 @@ export default function AgentPickerModal({ type, text, onClose }) {
     setNoAgentsAlert(false);
     setAddingWebId(true);
     const emptyThing = createThing();
+    const { dataset: addressBookDataset } = addressBook;
     const newItem = {
-      dataset: placeholderDataset,
       thing: emptyThing,
+      dataset: addressBookDataset,
     };
     setContactsArray([newItem, ...contactsArray]);
   };
 
+  const contactsForTable = selectedTabValue ? filteredContacts : contactsArray;
+
   const toggleCheckbox = (e, index, value) => {
     if (index === 0 && addingWebId) return null;
+    if (
+      !webIdsToDelete.includes(value) &&
+      webIdsInPermissions.includes(value)
+    ) {
+      setWebIdsToDelete([value, ...webIdsToDelete]);
+    } else if (
+      webIdsToDelete.includes(value) &&
+      webIdsInPermissions.includes(value)
+    ) {
+      setWebIdsToDelete(webIdsToDelete.filter((webId) => webId !== value));
+    }
     return e.target.checked
       ? setNewAgentsWebIds([value, ...newAgentsWebIds])
       : setNewAgentsWebIds(newAgentsWebIds.filter((webId) => webId !== value));
   };
-
   return (
     <div
       className={classes.paper}
       data-testid={TESTCAFE_ID_ADD_AGENT_PICKER_MODAL}
     >
-      <div className={classes.title}>{`${text} for ${resourceName}`}</div>
+      <div className={classes.title}>{`${editText} for ${resourceName}`}</div>
       <PolicyHeader type={type} />
       <div className={classes.tableContainer}>
         <div className={classes.tabsAndAddButtonContainer}>
           <AgentsTableTabs
             handleTabChange={handleTabChange}
             selectedTabValue={selectedTabValue}
             className={classes.modalTabsContainer}
+            tabsValues={{
+              all: "",
+              people: PERSON_CONTACT,
+              groups: GROUP_CONTACT,
+            }}
           />
           <AddWebIdButton onClick={handleAddRow} disabled={addingWebId} />
         </div>
         <AgentsSearchBar handleFilterChange={handleFilterChange} />
-        {contactsArray.length ? (
+        {!contacts && !error && (
+          <Container variant="empty">
```

Comment:
Looks like we should support a variant for the Loading Indicator?

Reply:
(More like a general question; if you agree, I'll create or add it to a ticket somewhere.)

Reply:
Yes, I just used it a second time so it's a good idea.